vertx-web/src/main/java/io/vertx/ext/web/impl/HttpServerRequestWrapper.java

```diff
@@ -149,6 +152,37 @@ public String query() {
     return query;
   }
 
+  @Override
+  public MultiMap params() {
+    if (!modified) {
+      return delegate.params();
+    }
+    if (params == null) {
+      params = MultiMap.caseInsensitiveMultiMap();
+      // if there is no query it's not really needed to parse it
+      if (query != null) {
+        QueryStringDecoder queryStringDecoder = new QueryStringDecoder(uri);
+        Map<String, List<String>> prms = queryStringDecoder.parameters();
+        if (!prms.isEmpty()) {
```

Comment:
Is this empty check mandatory? Without it I think it would exit out of the loop immediately as the entry set would be empty

Reply:
the `isEmpty` check can be optimized by the compiler as some implementations rely on verifying if the size is 0 and avoids any other work.